src/joint_state_listener.cpp

```diff
@@ -82,61 +164,76 @@ void JointStateListener::callbackFixedJoint(const ros::TimerEvent& e)
 
 void JointStateListener::callbackJointState(const JointStateConstPtr& state)
 {
-  if (state->name.size() != state->position.size()){
-    ROS_ERROR("Robot state publisher received an invalid joint state vector");
-    return;
-  }
+  /// continue processing of state-msgs to not collect old msgs in queue
+  if (update_ongoing)
```

Comment:
If you changed this variable to a mutex, rather than a bool, you should use `try_lock` here on the mutex variable instead.